templates/rust/src/client.rs.twig

```diff
@@ -0,0 +1,254 @@
+            .request(method_type, self.endpoint.join(&format!("{}{}", "v1", path)).unwrap());
+
+        match params {
+            Some(data) => {
+                let flattened_data = flatten(FlattenType::Normal(data.clone()), None);
+
+                // First flatten the data and feed it into a FormData
+                if content_type.starts_with("multipart/form-data") {
+                    let mut form = reqwest::blocking::multipart::Form::new();
+
+                    for (k, v) in flattened_data.clone() {
+                        match v {
+                            ParamType::Bool(data) => {
+                                form = form.text(k, data.to_string());
+                            }
+                            ParamType::String(data) => form = form.text(k, data),
+                            ParamType::FilePath(data) => form = form.file(k, data).unwrap(),
+                            ParamType::Number(data) => form = form.text(k, data.to_string()),
+                            // This shouldn't be possible due to the flatten function, so we won't handle this for now
+                            ParamType::Array(_data) => {
+                                //todo: Feed this back into a flatten function if needed
+                            },
+                            ParamType::Object(_data) => {
+                              // Same for this
+                            }
+                        }
+                    }
+                    request = request.multipart(form);
+                }
+
+                if content_type.starts_with("application/json") && method != "GET" {
+                    request = request.json(&data);
+                }
+
+                if method == "GET" {
+                    request = request.query(&queryize_data(flatten(FlattenType::Normal(data), None)));
+                }
+            }
+            None => {}
+        }
+
+        request = request.headers(request_headers);
+
+        match request.send() {
+            Ok(data) => {
+                if data.status().is_success() {
+                    Ok(data)    
+                } else {
+                    // Format error
+                    Err(match data.json() {
+                        Ok(data) => data,
+                        Err(err) => {
+                            {{spec.title | caseUcfirst}}Exception::new(format!("A error occoured. ERR: {}, This could either be a connection error or an internal Appwrite error. Please check your Appwrite instance logs. ", err), 0, "".to_string())
+                        }
+                    })
+                }
+            },
+            Err(err) => {
+                // Throw {{spec.title | caseUcfirst}} Exception
+                Err({{spec.title | caseUcfirst}}Exception::new(format!("A error occoured. ERR: {}, This could either be a connection error or an internal Appwrite error. Please check your Appwrite instance logs. ", err), 0, "".to_string()))
```

Comment:
May be only throw the underlying err, without extra message as we do with other SDKs